### PR TITLE
added varities dropdown to change pokemon

### PR DESCRIPTION
**Pull request by ethanjurman**
TODO: figure out why changing pokemon ("Charizard" -> "Muk") breaks it 😢 

image

---

image

---

Also now we can load pokemon without having to load their crazy name (before you'd have to load deoxys-normal, but now deoxys loads and we check `is_default` to load the right one)

image

### CODE REVIEW

File: src/hooks/usePokemon/usePokemon.js [hunk 1]
```diff
@@ -11,17 +11,74 @@ const usePokemonName = () => {
...
+    const pokemonDefaultVariation = pokemonSpeciesObject.varieties.find(({is_default}) => is_default)
+    const pokemonObject = await PokeAPI.getPokemonByName(pokemonDefaultVariation.pokemon.name)
     setPokemonMap({ ...pokemonMap, [pokemonName]: pokemonObject})
   }, [pokemonName])
+
+  useEffect(async () => {
+    // if (pokemonMap[pokemonName]) {
+    //   return pokemonMap[pokemonName]
+    // }
+    // setPokemonMap({ ...pokemonMap, [pokemonName]: "LOADING"})
```
Comment (JRJurman) [inline src/hooks/usePokemon/usePokemon.js]: comments?

File: src/hooks/usePokemon/usePokemon.js [hunk 2]
```diff
@@ -11,17 +11,74 @@ const usePokemonName = () => {
...
+    const varient = event.target.value;
+    setPokemonVariation({
+      ...pokemonVariation, [index]: varient
+    })
+  }
+
+  useEffect(() => {
+    if (pokemonVariation[0]) {
+      setPokemonVariation({
+        ...pokemonVariation, 0: null // instead of 0, reset index of pokemonName when it's indexed
```
Comment (JRJurman) [inline src/hooks/usePokemon/usePokemon.js]: maybe move this comment up, or better explain what this useEffect is for?

File: src/index.js [hunk 3]
```diff
@@ -27,6 +29,7 @@ const home = () => {
       <PokeImage />
       <NameInput />
       <PokemonTypes />
+      <PokemonVarietySelect index=${0} />
```
Comment (JRJurman) [inline src/index.js]: if this should be a default value, maybe set that in the component itself, 
```
export default ({index = 0}) => {
```

File: src/hooks/usePokemon/usePokemon.js (hunk 1 above)
Comment (ethanjurman) [reply src/hooks/usePokemon/usePokemon.js]: I meant to remove these...

File: src/index.js (hunk 3 above)
Comment (ethanjurman) [reply src/index.js]: this is part of the assumption that each pokemon will (soon) have an index (and you'll have pokemon set of 0 through 5). It should not be a default value.

File: src/hooks/usePokemon/usePokemon.js (hunk 2 above)
Comment (ethanjurman) [reply src/hooks/usePokemon/usePokemon.js]: It's essentially for clearing your variation choice if you change your pokemonName (effectively you want a different pokemon).

I don't know where the best place to put that info is.... or if I can label useEffects

Comment (JRJurman) [reply src/hooks/usePokemon/usePokemon.js]: just a comment above the useEffect would be good

File: src/index.js (hunk 3 above)
Comment (JRJurman) [reply src/index.js]: ah, index is probably a bad variable name then... don't worry about this for now

**Pull request closed without merging**